reanahub/reana-client-go
# cli: add swagger client and filter option to list command

- Added automatically generated API client (https://github.com/go-swagger/go-swagger)
- Generated API client files with the updated openapi specification (https://github.com/reanahub/reana-commons/pull/355)
- Updated all available commands to use swagger api client
- Added `filter` option to the `list` command

To test:
- `go mod tidy`
- check if `ping` and `info` commands still works as expected
- try to play with `list` command `filter` option. e.g. `go run . list --filter name=roofit --filter status=running`
